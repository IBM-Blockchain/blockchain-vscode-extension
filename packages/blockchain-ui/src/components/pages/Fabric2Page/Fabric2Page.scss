@import '@carbon/layout/scss/_spacing';

#root {
    .fabric2-heading-container {
        margin-bottom: $spacing-09;

        .fabric2-icon-and-title {
            display: flex;
            align-items: center;
            margin-bottom: $spacing-07;
    
            .fabric2-rocket {
                height: 32px;
                margin-right: $spacing-05;
            }
        }
    }

    .fabric2-content-container {
        margin-bottom: $spacing-09;

        h4 {
            margin-bottom: $spacing-05;
        }

        .bx--accordion__content {
            p {
                margin-bottom: $spacing-05;

                &:last-of-type {
                    margin-bottom: 0px;
                }
            }

            ol {
                margin-left: $spacing-09;
                margin-bottom: $spacing-05;
            }
        }
    }

    .fabric2-additional-info-container {
        h4 {
            margin-bottom: $spacing-05;
        }
    }
}
