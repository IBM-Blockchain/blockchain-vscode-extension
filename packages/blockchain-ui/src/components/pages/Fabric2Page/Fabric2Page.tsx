import React from 'react';
import { Accordion, AccordionItem, ListItem, Link, OrderedList } from 'carbon-components-react';
import CommandLink from '../../elements/CommandLink/CommandLink';
import fabric2Rocket from '../../../resources/rocket.svg';
import './Fabric2Page.scss';
import { ExtensionCommands } from '../../../ExtensionCommands';

function Fabric2Page(): JSX.Element {
    return (
        <>
            <div className='fabric2-heading-container'>
                <div className='fabric2-icon-and-title'>
                    <img src={fabric2Rocket} alt='' className='fabric2-rocket' id='fabric2-rocket'></img>
                    <h3>Find out what's new with Fabric v2.0</h3>
                </div>
                <p>Congratulations on upgrading to Fabric v2.0! This update brings a new smart contract lifecycle, as well as new features. Learn about how this update will affect your work flow below</p>
            </div>
            <div className='fabric2-content-container'>
                <h4>What's new?</h4>
                <Accordion>
                    <AccordionItem title='Deploy smart contracts with ease' open>
                        <p>Smart contracts are now deployed using a new smart contract lifecycle. This deployment view introduces you to the Fabric v2.0 smart contract lifecycle through a series of easy-to-follow, convenient steps, aided by some information about how Fabric v2.0 smart contract deployment works underneath this flow.</p>
                        <p>We recommend you following the updated <CommandLink linkContents='A3: Deploying a smart contract' commandName={ExtensionCommands.OPEN_TUTORIAL_PAGE} commandData={['Basic tutorials', 'A3: Deploying a smart contract']}/> tutorial in the <CommandLink linkContents='Tutorial gallery' commandName={ExtensionCommands.OPEN_TUTORIAL_GALLERY}/> to learn how to access and use this deployment view.</p>
                    </AccordionItem>
                    <AccordionItem title='New smart contract lifecycle'>
                        <p>The smart contract lifecycle now has the following steps;</p>
                        <OrderedList>
                            <ListItem>Package your smart contract</ListItem>
                            <ListItem>Install the smart contract on your peers</ListItem>
                            <ListItem>Approve a generated smart contract definition</ListItem>
                            <ListItem>Commit the smart contract to the channel</ListItem>
                        </OrderedList>
                        <p>This allows for decentralised governance of smart contracts.</p>
                        <p>Don't worry about memorising this process as the deployment view makes it easy for you to deploy a smart contract without needing to know the steps above. Learn more about the new smart contract lifecycle and further Fabric v2.0 changes in the Fabric documentation <Link href='https://hyperledger-fabric.readthedocs.io/en/release-2.0/whatsnew.html'>here</Link>.</p>
                    </AccordionItem>
                </Accordion>
            </div>
            <div className='fabric2-additional-info-container'>
                <h4>Additional information</h4>
                <p>If you are running a production environment at v1.4, find the latest bug fixes for this <Link href='https://github.com/IBM-Blockchain/blockchain-vscode-extension/releases'>here</Link>.</p>
            </div>
        </>
    );
}

export default Fabric2Page;
