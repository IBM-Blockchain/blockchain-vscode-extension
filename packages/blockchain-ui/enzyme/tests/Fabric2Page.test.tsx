import React from 'react';
import renderer from 'react-test-renderer';
import chai from 'chai';
import sinonChai from 'sinon-chai';

import Fabric2Page from '../../src/components/pages/Fabric2Page/Fabric2Page';

chai.should();
chai.use(sinonChai);

describe('Fabric2Page component', () => {
    it('should render the expected snapshot', () => {
        const component: any = renderer
            .create(<Fabric2Page/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
