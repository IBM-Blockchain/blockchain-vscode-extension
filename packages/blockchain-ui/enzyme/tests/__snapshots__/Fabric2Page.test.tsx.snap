// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fabric2Page component should render the expected snapshot 1`] = `
Array [
  <div
    className="fabric2-heading-container"
  >
    <div
      className="fabric2-icon-and-title"
    >
      <img
        alt=""
        className="fabric2-rocket"
        id="fabric2-rocket"
        src="test-file-stub"
      />
      <h3>
        Find out what's new with Fabric v2.0
      </h3>
    </div>
    <p>
      Congratulations on upgrading to Fabric v2.0! This update brings a new smart contract lifecycle, as well as new features. Learn about how this update will affect your work flow below
    </p>
  </div>,
  <div
    className="fabric2-content-container"
  >
    <h4>
      What's new?
    </h4>
    <ul
      className="bx--accordion bx--accordion--end"
    >
      <li
        className="bx--accordion__item bx--accordion__item--active"
        onAnimationEnd={[Function]}
      >
        <button
          aria-expanded={true}
          className="bx--accordion__heading"
          onClick={[Function]}
          onKeyDown={[Function]}
          title="Expand/Collapse"
          type="button"
        >
          <svg
            aria-label="Expand/Collapse"
            className="bx--accordion__arrow"
            focusable="false"
            height={16}
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="11,8 6,13 5.3,12.3 9.6,8 5.3,3.7 6,3"
            />
          </svg>
          <div
            className="bx--accordion__title"
          >
            Deploy smart contracts with ease
          </div>
        </button>
        <div
          className="bx--accordion__content"
        >
          <p>
            Smart contracts are now deployed using a new smart contract lifecycle. This deployment view introduces you to the Fabric v2.0 smart contract lifecycle through a series of easy-to-follow, convenient steps, aided by some information about how Fabric v2.0 smart contract deployment works underneath this flow.
          </p>
          <p>
            We recommend you following the updated 
            <a
              className="bx--link"
              href=""
              onClick={[Function]}
            >
              A3: Deploying a smart contract
            </a>
             tutorial in the 
            <a
              className="bx--link"
              href=""
              onClick={[Function]}
            >
              Tutorial gallery
            </a>
             to learn how to access and use this deployment view.
          </p>
        </div>
      </li>
      <li
        className="bx--accordion__item"
        onAnimationEnd={[Function]}
      >
        <button
          aria-expanded={false}
          className="bx--accordion__heading"
          onClick={[Function]}
          onKeyDown={[Function]}
          title="Expand/Collapse"
          type="button"
        >
          <svg
            aria-label="Expand/Collapse"
            className="bx--accordion__arrow"
            focusable="false"
            height={16}
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="11,8 6,13 5.3,12.3 9.6,8 5.3,3.7 6,3"
            />
          </svg>
          <div
            className="bx--accordion__title"
          >
            New smart contract lifecycle
          </div>
        </button>
        <div
          className="bx--accordion__content"
        >
          <p>
            The smart contract lifecycle now has the following steps;
          </p>
          <ol
            className="bx--list--ordered"
          >
            <li
              className="bx--list__item"
            >
              Package your smart contract
            </li>
            <li
              className="bx--list__item"
            >
              Install the smart contract on your peers
            </li>
            <li
              className="bx--list__item"
            >
              Approve a generated smart contract definition
            </li>
            <li
              className="bx--list__item"
            >
              Commit the smart contract to the channel
            </li>
          </ol>
          <p>
            This allows for decentralised governance of smart contracts.
          </p>
          <p>
            Don't worry about memorising this process as the deployment view makes it easy for you to deploy a smart contract without needing to know the steps above. Learn more about the new smart contract lifecycle and further Fabric v2.0 changes in the Fabric documentation 
            <a
              className="bx--link"
              href="https://hyperledger-fabric.readthedocs.io/en/release-2.0/whatsnew.html"
            >
              here
            </a>
            .
          </p>
        </div>
      </li>
    </ul>
  </div>,
  <div
    className="fabric2-additional-info-container"
  >
    <h4>
      Additional information
    </h4>
    <p>
      If you are running a production environment at v1.4, find the latest bug fixes for this 
      <a
        className="bx--link"
        href="https://github.com/IBM-Blockchain/blockchain-vscode-extension/releases"
      >
        here
      </a>
      .
    </p>
  </div>,
]
`;
